src/Section/Tags.elm

```diff
@@ -0,0 +1,215 @@
+
+                updatedModelWithCmd =
+                    ( updatedModel
+                    , mappedCmd
+                    , tags
+                    )
+            in
+            case tagParentMsg of
+                FormTag.NoUpdate ->
+                    updatedModelWithCmd
+
+                FormTag.SubmittedForm submittedTag ->
+                    ( updatedModel
+                    , mappedCmd
+                    , tags ++ [ submittedTag ]
+                    )
+
+        FormMsg FormTag.Edit formMsg ->
+            editTagFormModel
+                |> Maybe.map
+                    (\( editTagFormModel_, originalTag ) ->
+                        let
+                            ( updatedEditTagFormModel, tagCmd, tagParentMsg ) =
+                                FormTag.update tags formMsg editTagFormModel_
+
+                            mappedCmd =
+                                Cmd.map (FormMsg FormTag.Edit) tagCmd
+
+                            hiddenEditTagFormModel =
+                                Model { modelData | editTagFormModel = Nothing }
+                        in
+                        case tagParentMsg of
+                            FormTag.NoUpdate ->
+                                ( Model { modelData | editTagFormModel = Just ( updatedEditTagFormModel, originalTag ) }
+                                , mappedCmd
+                                , tags
+                                )
+
+                            FormTag.SubmittedForm submittedTag ->
+                                ( hiddenEditTagFormModel
+                                , mappedCmd
+                                , updateTagInList
+                                    { originalTag = originalTag
+                                    , updatedTag = submittedTag
+                                    }
+                                    tags
+                                )
+                    )
+                |> Maybe.withDefault
+                    ( model
+                    , Cmd.none
+                    , tags
+                    )
+
+
+updateTagInList : { originalTag : Tag.Model, updatedTag : Tag.Model } -> Tag.Tags -> Tag.Tags
+updateTagInList { originalTag, updatedTag } =
+    List.map
+        (\tag ->
+            if Tag.isSameTagName tag originalTag then
+                updatedTag
+
+            else
+                tag
+        )
+
+
+view : Tag.Tags -> Model -> Html Msg
+view tags (Model { addTagFormModel, editTagFormModel }) =
+    Html.div [ Attributes.class "w-96" ]
+        [ tagsView tags
+        , Html.h1 [ Attributes.class "text-center mt-5" ] [ Html.text "Create new tag" ]
+        , Html.div [ Attributes.class "mt-2" ]
+            [ FormTag.view addTagFormModel
+                |> Html.map (FormMsg FormTag.Add)
+            ]
+        , editTagFormModel
+            |> Maybe.map modalView
+            |> Maybe.withDefault (Html.text "")
+        ]
```

Comment:
Were there any particular considerations for splitting creation and editing into two completely different UIs?
For example, what lead you to the decision to use a modal for one but not for the other?

Reply:
I understand. Thanks! =)